src/aspire/basis/basis.py

```diff
@@ -174,6 +176,10 @@ def expand(self, x):
             those first dimensions of `x`.
 
         """
+
+        if isinstance(x, Image) or isinstance(x, Volume):
```

Comment:
Probably ASPIRE can embrace the classes/api more in the future now that most components exist.

I don't know that this change needs to go in this PR, but you will need it for converting `evaluate_t` to `expand`. (This is how I found it iirc).

Reply:
No it was more that I encountered this bug during #576 trying to feed `Image`s into `expand`, stupidly thinking that since `evaluate_t` allows it, `expand` would as well…

Reply:
So yes, let's keep it in here.